sql.go

```diff
@@ -58,22 +100,115 @@ func contains(array []string, element string) bool {
 	return false
 }
 
+// defineConstants defines the constants that can be used in the JS code.
+func (sql *SQL) defineConstants() {
+	runtime := sql.vu.Runtime()
+	mustAddProp := func(name string, val interface{}) {
+		err := sql.exports.DefineDataProperty(
+			name, runtime.ToValue(val), sobek.FLAG_FALSE, sobek.FLAG_FALSE, sobek.FLAG_TRUE,
+		)
+		if err != nil {
+			common.Throw(runtime, err)
+		}
+	}
+
+	// TLS versions
+	mustAddProp("TLS_1_0", netext.TLS_1_0)
+	mustAddProp("TLS_1_1", netext.TLS_1_1)
+	mustAddProp("TLS_1_2", netext.TLS_1_2)
+	mustAddProp("TLS_1_3", netext.TLS_1_3)
+}
+
+type TLSConfig struct {
+	EnableTLS             bool   `json:"enableTLS"`
+	InsecureSkipTLSverify bool   `json:"insecureSkipTLSverify"`
+	MinVersion            string `json:"minVersion"`
+	CAcertFile            string `json:"caCertFile"`
+	ClientCertFile        string `json:"clientCertFile"`
+	ClientKeyFile         string `json:"clientKeyFile"`
+}
+
+// LoadTLS loads the TLS configuration for the SQL module.
+func (sql *SQL) LoadTLS(params map[string]interface{}) {
+	runtime := sql.vu.Runtime()
+	var tlsConfig *TLSConfig
+	if b, err := json.Marshal(params); err != nil {
+		common.Throw(runtime, err)
+	} else {
+		if err := json.Unmarshal(b, &tlsConfig); err != nil {
+			common.Throw(runtime, err)
+		}
+	}
+	sql.tlsConfig = *tlsConfig
+}
+
 // Open establishes a connection to the specified database type using
 // the provided connection string.
-func (*SQL) Open(database string, connectionString string) (*dbsql.DB, error) {
+func (sql *SQL) Open(database string, connectionString string) (*dbsql.DB, error) {
 	supportedDatabases := []string{"mysql", "postgres", "sqlite3", "sqlserver", "azuresql", "clickhouse"}
 	if !contains(supportedDatabases, database) {
 		return nil, fmt.Errorf("database %s is not supported", database)
 	}
 
+	if database == "mysql" && sql.tlsConfig.EnableTLS {
+		tlsConfigKey := "custom"
+		if err := registerTLS(tlsConfigKey, sql.tlsConfig); err != nil {
+			return nil, err
+		}
+		connectionString = prefixConnectionString(connectionString, tlsConfigKey)
+	}
+
 	db, err := dbsql.Open(database, connectionString)
 	if err != nil {
 		return nil, err
 	}
-
 	return db, nil
 }
 
+// prefixConnectionString prefixes the connection string with the TLS configuration key.
+func prefixConnectionString(connectionString string, tlsConfigKey string) string {
+	var separator string
+	if strings.Contains(connectionString, "?") {
+		separator = "&"
+	} else {
+		separator = "?"
+	}
+	return fmt.Sprintf("%s%stls=%s", connectionString, separator, tlsConfigKey)
+}
+
+// registerTLS loads the ca-cert and registers the TLS configuration with the MySQL driver.
+func registerTLS(tlsConfigKey string, tlsConfig TLSConfig) error {
+	rootCAs := x509.NewCertPool()
+	{
+		pem, err := os.ReadFile(tlsConfig.CAcertFile)
+		if err != nil {
+			return err
+		}
+		if ok := rootCAs.AppendCertsFromPEM(pem); !ok {
+			return fmt.Errorf("failed to append PEM")
+		}
+	}
```

Comment:
what do we need the extra `{}` block for?

Reply:
Done. Thank you.